spec/models/post_spec.rb

```diff
@@ -336,36 +337,6 @@ def post_with_body(body, user = nil)
       expect(post_one_image).to be_valid
     end
 
-    it "doesn't allow more than `min_trust_to_post_embedded_media`" do
-      SiteSetting.min_trust_to_post_embedded_media = 4
-      post_one_image.user.trust_level = 3
-      expect(post_one_image).not_to be_valid
-    end
-
-    it "doesn't allow more than `min_trust_to_post_embedded_media` in a quote" do
-      SiteSetting.min_trust_to_post_embedded_media = 4
-      post_one_image.user.trust_level = 3
-      expect(post_image_within_quote).not_to be_valid
-    end
-
-    it "doesn't allow more than `min_trust_to_post_embedded_media` in code" do
-      SiteSetting.min_trust_to_post_embedded_media = 4
-      post_one_image.user.trust_level = 3
-      expect(post_image_within_code).not_to be_valid
-    end
-
-    it "doesn't allow more than `min_trust_to_post_embedded_media` in pre" do
-      SiteSetting.min_trust_to_post_embedded_media = 4
-      post_one_image.user.trust_level = 3
-      expect(post_image_within_pre).not_to be_valid
-    end
-
-    it "doesn't allow more than `min_trust_to_post_embedded_media`" do
-      SiteSetting.min_trust_to_post_embedded_media = 4
-      post_one_image.user.trust_level = 4
```

Comment:
I'm wondering if we should rig `User#trust_level=` up to proxy to `#change_trust_level!`. WDYT?

Reply:
I think it's probably too tricky at this point...also it violates the whole rails model thing where you can set a bunch of attributes and then do `.save!` . Sadly the ship has sailed :(

There are a lot of these older written specs that just modify attrs instead of going to the DB; I'm sure it was fine at the time, in practice I find this stuff so annoying and would universally prefer that the actual DB state was being changed.